shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -563,19 +585,23 @@
     int64_t view_id,
     int64_t overlay_id) {
   auto layer = layer_pool_->GetLayer(gr_context, ios_context);
+
+  auto overlay_view_wrapper = layer->overlay_view_wrapper.get();
   auto screenScale = [UIScreen mainScreen].scale;
-  // Set the size of the overlay UIView.
-  layer->overlay_view.get().frame = CGRectMake(rect.x() / screenScale,      //
-                                               rect.y() / screenScale,      //
-                                               rect.width() / screenScale,  //
-                                               rect.height() / screenScale  //
-  );
-  // Set a unique view identifier, so the overlay can be identified in unit tests.
-  layer->overlay_view.get().accessibilityIdentifier =
+  // Set the size of the overlay view wrapper.
+  // This wrapper view masks the overlay view.
+  overlay_view_wrapper.frame = CGRectMake(rect.x() / screenScale, rect.y() / screenScale,
+                                          rect.width() / screenScale, rect.height() / screenScale);
+  // Set a unique view identifier, so the overlay wrapper can be identified in unit tests.
+  overlay_view_wrapper.accessibilityIdentifier =
       [NSString stringWithFormat:@"platform_view[%lld].overlay[%lld]", view_id, overlay_id];
 
-  std::unique_ptr<SurfaceFrame> frame =
-      layer->surface->AcquireFrame(SkISize::Make(rect.width(), rect.height()));
+  auto overlay_view = layer->overlay_view.get();
```

Comment:
superfluous "auto"

Reply:
I heard @chinmaygarde likes auto :-)

Reply:
Doesn't matter what @chinmaygarde likes, the google style guide is clear.  It's the same constitution we all adhere to including him.

Reply:
In this case in particular, `overlay_view` refers to a view. However, I acknowledge the [style guide](https://google.github.io/styleguide/cppguide.html#Type_deduction) has left this to the author's judgment who may be biased. For example, I think someone can infer that this is a UIView because of the word `view` in the variable, and the fact that the properties being set are very used in iOS development.

I don't have strong preference. Generally speaking, I like type inference. Although, I'm happy to change it if it helps a new contributor.

Reply:
I'd personally don't feel strongly about it either, but the style guide is clear.  I learned years ago, everyone is happier at Google if we just follow the guide.  There is too much overhead to debating this stuff all the time or trying to figure out what the unwritten style is =)

Reply:
I heard what you are saying. My rule of thumb is that if the property name carries the meaning of it and type, there is no need to specify the type. Someone can do `layer->overlay_view.get().frame = CGMakeRect(...)` instead of first setting `layer->overlay_view.get()` as local variable with an explicit type.

Reply:
> Use type deduction only if it makes the code clearer to readers who aren't familiar with the project, or if it makes the code safer. Do not use it merely to avoid the inconvenience of writing an explicit type.

I would say this doesn't make the code clearer because someone who isn't familiar with the code doesn't know the type of `layer->overlay_view` and it doesn't make it any safer.

> if the property name carries the meaning of it and type, there is no need to specify the type

The property doesn't specify the type, it says `view` but that's not the same thing as a `UIView`.